code/components/extra-natives-five/src/ModelListNatives.cpp

```diff
@@ -0,0 +1,63 @@
+#include <StdInc.h>
+#include <Hooking.h>
+
+#include <ScriptEngine.h>
+#include <ScriptSerialization.h>
+
+static uint32_t(*initVehicleArchetype)(const char* name, bool a2, unsigned int a3);
+
+static void(*g_origArchetypeDtor)(fwArchetype* at);
+
+// Use a map of hashes and names for optimization purposes
+// We just hash it on registration instead of needing to rehash it when checking for it every deregistration
+static std::unordered_map<uint32_t, std::string> g_vehicles;
+
+static uint32_t initVehicleArchetype_stub(const char* name, bool a2, unsigned int a3)
+{
+	uint32_t hash = HashString(name);
+	if (g_vehicles.find(hash) == g_vehicles.end())
+	{
+		g_vehicles.insert({ hash, name });
+	}
+	return initVehicleArchetype(name, a2, a3);
+}
+
+static void ArchetypeDtorHook(fwArchetype* at)
+{
+	if (at->assetType == 1)
```

Comment:
This one was done just as a measure to optimize to avoid running on ALL unregistered archetypes (which seems to happen during runtime a fair bit iirc?). Theres no specific one for vehicles so this seemed preferable still just for some minor optimization.